artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -1269,7 +1269,7 @@ private AddressInfo checkAddress(RoutingContext context, SimpleString address) t
 
    Bindings simpleRoute(SimpleString address, RoutingContext context, Message message, AddressInfo addressInfo) throws Exception {
       Bindings bindings = addressManager.getBindingsForRoutingAddress(address);
-      if (bindings == null && context.getServerSession() != null) {
+      if ((bindings == null || !bindings.contains(LocalQueueBinding.class)) && context.getServerSession() != null) {
```

Comment:
simpleRoute is a hot method. This would cause an iteration on bindings to verify if hasLocal or not. 

I'm amending your commit to have the attribute cached.


The only place this method was called (which was recently introduced) was during a queue creation.. and there it wouldn't be as hot as this one.